### DISCUSSION TITLE
Kconfig symbol selected in board revision config causes warnings when building mcuboot

### DISCUSSION THREAD

**Discussion by cdwilson** [zephyrproject-rtos/zephyr#67808] [Q&A]
I'm working on a board that uses [Zephyr board revisions](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#multiple-board-revisions) to handle minor differences between the hardware revs.

The "Rev A" of the board was built with NRF9160-SIBA-R7 (NB-IoT only), but future revisions of the board were built with NRF9160-SICA-B1A-R7 (LTE-M/NB-IoT/GNSS).

To make sure the firmware uses NB-IoT for the Rev A board, I have added `boards/arm/<<board name redacted>>/<<board name redacted>>_ns_A.conf` with the following config:

```
# Select NB-IoT system mode when using the LTE Link Control driver:
CONFIG_LTE_NETWORK_MODE_NBIOT=y
CONFIG_LTE_NETWORK_USE_FALLBACK=n
```

However, when I build the firmware (`west build -p -b <<board name redacted>>_ns@A`), I get the following warning when building the mcuboot child image (building Zephyr app image builds without warnings):

```
warning: The choice symbol LTE_NETWORK_MODE_NBIOT (defined at <<redacted>>/nrf/lib/lte_link_control/Kconfig:166) was selected (set =y), but no symbol ended up as
the choice selection. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LTE_NETWORK_MODE_NBIOT and/or look up
LTE_NETWORK_MODE_NBIOT in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.
```

Is there a recommended way to force the NB-IoT mode only for board Rev A (`<<board name redacted>>_ns@A`) that will not result in this warning when building mcuboot?

**Comment by nordicjm** [comment]
You would use an overlay for your application instead

**Reply by cdwilson** [reply]
> You would use an overlay for your application instead

@nordicjm could you explain this in a bit more detail?

I forgot this repo was public when I posted the original question, so you can see the actual config file for the Revision A board here:

https://github.com/golioth/golioth-zephyr-boards/blob/cdwilson/aludel-elixir-v1-support/boards/arm/aludel_elixir_v1/aludel_elixir_v1_A.conf

To provide a bit more context, this board is defined in a Zephyr module that is pulled in as a dependency for multiple Zephyr apps  (see [example](https://github.com/golioth/reference-design-template/blob/b6e2182a3986c257afa11ea85a00378fcd5f500c/west.yml#L33)). I would like to be able to define this special NB-IoT configuration for the Rev A board once in the module, rather than having to configure something for each app.

The issue appears to be that both the app and mcuboot are pulling in the `aludel_elixir_v1_A.conf`, but `CONFIG_LTE_NETWORK_MODE_NBIOT=y` is causing issues when building the mcuboot child image only (app image builds fine). I'm wondering if there is a way to prevent these symbols from being defined when the mcuboot child image is built.
